src/components/Todo.tsx

```diff
@@ -62,11 +96,11 @@ export default function Todos() {
     return 0;
   });
 
-  const indexOfLastTodo = currentPage * todosPerPage;
-  const indexOfFirstTodo = indexOfLastTodo - todosPerPage;
+  const indexOfLastTodo = currentPage * TodosPerPage;
+  const indexOfFirstTodo = indexOfLastTodo - TodosPerPage;
   const currentTodos = filteredTodos.slice(indexOfFirstTodo, indexOfLastTodo);
 
-  const paginate = (pageNumber: number) => setCurrentPage(pageNumber);
+  const paginate = (pageNumber: number) => dispatch({ type: ActionTypes.SET_CURRENT_PAGE, payload: pageNumber });
```

Comment:
You can also make use of action creator functions for eg-
here you can have a function defined already which is
setCurrentPageAction = (pageNumber) => ({
type: ActionTypes.SET_CURRENT_PAGE, payload: pageNumber
})

& then while dispatching action you can do dispatch(setCurrentPageAction(pageNumber))